cpp/tests/core/SYCLUtils.cpp

```diff
@@ -34,12 +37,55 @@
 namespace open3d {
 namespace tests {
 
-class SYCLUtilsPermuteDevices : public PermuteDevices {};
-INSTANTIATE_TEST_SUITE_P(SYCLUtils,
-                         SYCLUtilsPermuteDevices,
-                         testing::ValuesIn(PermuteDevices::TestCases()));
+TEST(SYCLUtils, SYCLDemo) { core::sycl_utils::SYCLDemo(); }
+
+TEST(SYCLUtils, PrintAllSYCLDevices) {
```

Comment:
running test with `"*SYCL*"` only runs a single test for me. 
other tests are not running. 
Also, I have an integrated gpu, so should I expect SYCL to run on both CPU and integrated GPU ?
```
build(master)$ ./bin/tests --gtest_filter="*SYCL*"
[Open3D INFO] CompilerInfo: C++ 17, IntelLLVM 2022.1.0, CUDA disabled.
[Open3D INFO] CPUInfo: 8 cores, 16 threads.
[Open3D INFO] ISAInfo: AVX512SKX instruction set used in ISPC code.
Note: Google Test filter = *SYCLUtils*
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from SYCLUtils/SYCLUtilsPermuteDevices
[ RUN      ] SYCLUtils/SYCLUtilsPermuteDevices.SYCLDemo/0
[Open3D INFO] SYCLDemo passed!
[       OK ] SYCLUtils/SYCLUtilsPermuteDevices.SYCLDemo/0 (116 ms)
[----------] 1 test from SYCLUtils/SYCLUtilsPermuteDevices (116 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (116 ms total)
[  PASSED  ] 1 test.
```